File: 275. H-Index II/Solution.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        int l = 0, r = citations.size() - 1, n = citations.size();
        while ( l <= r ) {
            int mid = ( l + r ) / 2;
            if ( citations[mid] >= n-mid) { r = mid - 1; }
            else { l = mid + 1; }
        }
        return n - l;
    }
};

File: 275. H-Index II/Solution.py
class Solution(object):
    def hIndex(self, citations):
        """
        :type citations: List[int]
        :rtype: int
        """
        l, r, n = 0, len(citations)-1, len(citations)
        while l <= r:
            mid = (l + r) // 2
            if citations[mid] >= n - mid:
                r = mid - 1
            else:
                l = mid + 1
        return n - l

